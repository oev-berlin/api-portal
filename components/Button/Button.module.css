.button {
  font-weight: 700;
  border: 0;
  border-radius: 3em;
  cursor: pointer;
  display: inline-block;
  line-height: 1;
}

.button-primary {
  color: white;
  background-color: rgb(30 167 253);
}

.button-secondary {
  color: rgb(51 51 51);
  background-color: transparent;
  box-shadow: rgb(0 0 0 / 15%) 0 0 0 1px inset;
}

.button-small {
  font-size: 12px;
  padding: 10px 16px;
}

.button-medium {
  font-size: 14px;
  padding: 11px 20px;
}

.button-large {
  font-size: 16px;
  padding: 12px 24px;
}
