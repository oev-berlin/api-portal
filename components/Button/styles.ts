import styled from 'styled-components';
import { Button } from '@mui/material';

export const BackButton = styled(Button)`
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 1rem;
  display: block;
`;
