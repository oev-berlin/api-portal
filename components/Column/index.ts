import { Column } from './Column';
import { ColumnProps } from './interfaces';

export { Column };
export type { ColumnProps };
