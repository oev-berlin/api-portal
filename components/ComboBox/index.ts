import ComboBox from './ComboBox';
import { ComboBoxProps } from './interfaces';

export { ComboBox };
export type { ComboBoxProps };
