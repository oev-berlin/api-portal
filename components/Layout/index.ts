import { Layout } from './Layout';
import { LayoutProps } from './interfaces';

export { Layout };
export type { LayoutProps };
