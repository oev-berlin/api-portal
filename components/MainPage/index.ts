export {CustomisedCard} from "./ProjectCard";
export { MainPage } from './MainPage';
