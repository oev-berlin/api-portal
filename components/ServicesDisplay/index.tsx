import { ServicesDisplay } from './ServicesDisplay';

export { ServicesDisplay };
