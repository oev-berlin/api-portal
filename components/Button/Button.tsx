import React from 'react';
import Link from 'next/link';
import ButtonProps from './interfaces';
import { BackButton } from './styles';

export const Button = ({ text }:ButtonProps) => (
  <Link href="/">
    <BackButton variant="contained"> { text } </BackButton>
  </Link>
);
