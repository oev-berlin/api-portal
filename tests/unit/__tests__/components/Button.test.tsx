import { render, cleanup } from '@testing-library/react';
import {axe, toHaveNoViolations} from 'jest-axe';
import React from 'react';
import { Button } from '../../../../components/Button';
import 'jest-styled-components';
import ButtonProps from "../../../../components/Button/interfaces";
import { act } from 'react-dom/test-utils';

expect.extend(toHaveNoViolations);

const setupComponent = ({ text, link }: ButtonProps ) => render(
  <Button text={text} link={link} />,
);

let screen;
describe('Button', () => {
  afterEach(() => {
    cleanup();
  });

  describe('UI Tests', () => {
    it('Button component should render without crashing', async () => {
      act(()=>{screen = setupComponent({text:"Back", link:"/"})});
      expect(screen.container).toBeDefined();
    });

    it('Card component (empty text) should render without crashing', async () => {
      act( ()=>{screen = setupComponent({text: "", link:"/"})});
      expect(screen.container).toBeDefined();
    });
    it('Card component (empty link) should render without crashing', async () => {
      act( ()=>{screen = setupComponent({text: "Back", link:""})});
      expect(screen.container).toBeDefined();
    });
    it('Card component (empty text and empty link) should render without crashing', async () => {
      act( ()=>{screen = setupComponent({text: "", link:""})});
      expect(screen.container).toBeDefined();
    });

    it('Card component should be enabled', async () => {
      act( ()=>{screen = setupComponent({text: "Back", link:"/"})});
      await screen
          .findByText(/back/i)
          .then((element) =>
              expect(element).not.toBeDisabled());
    });

    it('Button component should render correctly', async () => {
      act( ()=>{screen = setupComponent({text:"Back", link:"/"})});
      await screen
        .findByText(/back/i)
        .then((element) => expect(element).toBeInTheDocument());
    });
  });

  describe('Snapshots Tests', () => {
    it('should match a basic snapshot', () => {
      act( ()=>{screen  = setupComponent({text:"Back", link:"/"})});
      expect(screen.container).toMatchSnapshot();
    });

    it('should match a basic snapshot (empty text)', () => {
      act(()=>{ screen = setupComponent({text:"", link:"/"})});
      expect(screen.container).toMatchSnapshot();
    });
    it('should match a basic snapshot (empty link)', () => {
      act(()=>{ screen = setupComponent({text:"Back", link:""})});
      expect(screen.container).toMatchSnapshot();
    });
    it('should match a basic snapshot (empty text and empty link)', () => {
      act(()=>{ screen = setupComponent({text:"", link:""})});
      expect(screen.container).toMatchSnapshot();
    });
  });

  describe('Accessibility Tests', () => {
    it('should pass a basic accessibility test', async () => {
      act(()=>{ screen = setupComponent({ text:"Back", link:"/"})});
      const results = await axe(screen.container);
      expect(results).toHaveNoViolations();
    });
  });
});
