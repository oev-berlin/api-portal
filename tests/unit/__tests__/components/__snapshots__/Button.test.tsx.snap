// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshots Tests should match a basic snapshot (empty link) 1`] = `
.c0 {
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 1rem;
  display: block;
}

<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium c0 css-sghohy-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
     
    Back
     
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Button Snapshots Tests should match a basic snapshot (empty text and empty link) 1`] = `
.c0 {
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 1rem;
  display: block;
}

<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium c0 css-sghohy-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
     
    
     
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Button Snapshots Tests should match a basic snapshot (empty text) 1`] = `
.c0 {
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 1rem;
  display: block;
}

<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium c0 css-sghohy-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
     
    
     
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Button Snapshots Tests should match a basic snapshot 1`] = `
.c0 {
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 1rem;
  display: block;
}

<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium c0 css-sghohy-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
     
    Back
     
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
